dojo/tools/govulncheck/parser.py

```diff
@@ -0,0 +1,68 @@
+import json
+from itertools import groupby
+from itertools import islice
+import logging
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+SEVERITY = 'Info'
+
+
+class GovulncheckParser:
+
+    def get_scan_types(self):
+        return ["Govulncheck Scanner"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Govulncheck Scanner findings in JSON format."
+
+    @staticmethod
+    def get_location(data, node):
+        while data['Calls']['Functions'][str(node)]['CallSites'][0]['Parent'] != 1:
+            node = data['Calls']['Functions'][str(node)]['CallSites'][0]['Parent']
+        return [f"{x['Pos']['Filename']}:{x['Pos']['Line']}:{x['Pos']['Column']}" for x in
+                data['Calls']['Functions'][str(node)]['CallSites']]
+
+    @staticmethod
+    def get_version(data, node):
+        return data['Requires']['Modules'][str(node)]['Version']
+
+    def get_findings(self, scan_file, test):
+        findings = []
+        try:
+            data = json.load(scan_file)
+        except Exception as e:
+            logger.warning("Invalid JSON format. %r", e)
+            return findings
```

Comment:
you're still doing exception swallowing

Reply:
Maybe use something like: raise ValueError("Invalid JSON format") ?

Reply:
or simply not catching the exception.